src/catalog/catalog.cpp

```diff
@@ -168,6 +168,16 @@ void Catalog::Bootstrap() {
 // CREATE FUNCTIONS
 //===----------------------------------------------------------------------===//
 
+bool Catalog::CheckDatabaseExists(const std::string &database_name) {
+	auto pg_database = DatabaseCatalog::GetInstance();
+	auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+	auto txn = txn_manager.BeginTransaction();
```

Comment:
Instead of starting a txn inside Catalog, it would be better starting a new txn outside and pass it into the function. It would be more consistent with the form of other functions in Catalog.

Reply:
Right.